code/_helpers/unsorted.dm

```diff
@@ -1258,4 +1258,13 @@ var/list/FLOORITEMS = list(
 	else
 		return 0
 
+//Prevents robots dropping their modules.
+/proc/dropsafety(var/atom/movable/A)
```

Comment:
That's complicated code.
Isn't it better to rewrite 'unEquip(...)` and `drop_from_inventory(...)` for cyborgs like that (example)?
```
/mob/living/silicon/robot/unEquip(obj/item/I, var/atom/Target)
	if(istype(module_active, /obj/item/weapon/gripper))
		var/obj/item/weapon/gripper/G = module_active
		if(I == G || G.wrapped == I)
			return G.drop_to(Target)
	usr << SPAN_WARN("You can't unequip [I]!")
	return FALSE
```

Reply:
That's not complicated at all, it's like six lines.
Some inventory procs are already overridden, but not all and that's intentional

If you let the code get to the stage of attempting to unequip the item before checking things, it gets messier and more complex

Reply:
also, this proc also provides safety for rig modules

Reply:
>>> That's not complicated at all
This shouldn't be add to each proc where user can drop items?

Rig modules also should be handled in internal inventory-related proc. Don't you think so?

Reply:
also, i already overrode can_UnEquip which is far more useful and correct